docs/source/aggregation.rst

```diff
@@ -0,0 +1,187 @@
+where all the jobs present in the project are passed as arbitraty arguments (or ``*args``) and ``op2`` is a *normal operation*
+where only a single job is passed as a parameter.
+
+
+.. _types_of_aggregation:
+
+Types of Aggregation
+====================
+
+Currently, **signac-flow** allows users to aggregate jobs by:
+
+- Grouping them on state point key, an iterable of state point keys whose values define the
+  groupings, or an arbitrary callable of :class:`~signac.contrib.job.Job`.
+- Generating aggregates of a given size.
+- Using custom aggregator function when greater flexibility is needed.
+
+Group By
+---------
+
+:class:`~flow.aggregator.groupby` allows users to aggregate jobs by grouping them on
+state point key, an iterable of state point keys whose values define the groupings,
+or an arbitrary callable of :class:`~signac.contrib.job.Job`.
+
+.. code-block:: python
+
+    @aggregator.groupby('temperature')
+    @Project.operation
+    def op3(*jobs):
+        pass
+
+In the above example, the jobs will get aggregated based on the state point **temperature**.
+So, all the jobs having the same value of **temperature** in their state point will be aggregated together.
+
+Groups Of
+---------
+
+:class:`~flow.aggregator.groupsof` allows users to aggregate jobs by generating aggregates of a given size.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2)
+    @Project.operation
+    def op4(job1, job2):
+        pass
+
+In the above example, the jobs will get aggregated in groups of 2 and hence, only two jobs will
+be passed as parameters at once.
+
+.. note::
+
+    In case the number of jobs in the project is odd, there will be one aggregate containing only a single
+    job and hence users should be careful while defining the parameters for an *aggregate operation*.
+
+Sorting jobs for aggregation
+----------------------------
+
+**signac-flow** allows users to define the sorting order of jobs before creating the aggregates with the
+help of ``sort_by`` parameter and the sorting order can be defined with the help of ``sort_ascending`` parameter.
+
+.. code-block:: python
+
+    @aggregator.groupsof(2, sort_by='temperature', sort_ascending=False)
+    @Project.operation
+    def op5(job1, job2):
+        pass
+
+.. note::
+
+    In the above example, all the jobs will be sorted by the state point parameter ``temperature`` in descending
+    order and then be aggregated as groups of 2.
+
+Selecting jobs for aggregation
+------------------------------
+**signac-flow** allows users to selectively choose which jobs to pass into operation functions.
+
+.. code-block:: python
+
+    @aggregator(select=lambda job: job.sp.temperature > 0)
+    @Project.operation
+    def op6(job1, job2):
+        pass
+
+
+.. _aggregate_id:
+
+Aggregate ID
+============
+
+Similar to the concept of a job id, an aggregate id is a unique hash identifying an aggregate of jobs.
+The aggregate id is sensitive to the order of the jobs in the aggregate
```

Comment:
So what is the default order then? Should be documented here.

Reply:
Tried addressing here: https://github.com/glotzerlab/signac-docs/pull/122/files#diff-d053cd842dd6fe746188c4339762f7e336b763b94d513473d65900b9836a09f0R89